### PR TITLE
feat(ListView): Add compound expansion support

### LINKED ISSUE #174: ListView does not support compound expansion
The current implementation of ListView does not support compound expansion.
Pattern can be seen at http://www.patternfly.org/pattern-library/content-views/list-view/#example-code-3

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
**What**:
Adding compound expansion support to ListView

**Link to Storybook**:
[Storybook Link](https://jeff-phillips-18.github.io/patternfly-react/?knob-Stacked=false&selectedKind=ListView&selectedStory=List%20with%20Compound%20Expansion&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs)

**Patternfly Design**
http://www.patternfly.org/pattern-library/content-views/list-view/#compound-expansion

Fixes #174

### CODE REVIEW

File: src/components/ListView/ListViewItem.js [hunk 1]
```diff
@@ -58,15 +61,42 @@ class ListViewItem extends React.Component {
...
+            >
+              {children}
+            </ListViewGroupItemContainer>
+          </ListViewGroupItem>
+        );
+      } else {
+        return (
+          <ListViewGroupItem expanded={expanded} stacked={stacked} {...other}>
+            <ListViewGroupItemHeader
+              toggleExpanded={() => this.toggleExpanded()}
```
Comment (priley86) [inline src/components/ListView/ListViewItem.js]: i didn't catch this on previous reviews... can be simplified to `toggleExpanded={ this.toggleExpanded}`

File: src/components/ListView/ListViewItem.js [hunk 2]
```diff
@@ -103,7 +133,13 @@ ListViewItem.propTypes = {
...
-  checkboxInput: PropTypes.node
+  checkboxInput: PropTypes.node,
+  /** Flag to use compound expansion contents */
+  compoundExpand: PropTypes.bool,
+  /** Flag to show compound expansion contents */
+  compoundExpanded: PropTypes.bool,
+  /** Function triggered when compound expandable content is closed */
+  onCloseCompoundExpand: PropTypes.func
 };
 ListViewItem.defaultProps = {
```
Comment (priley86) [inline src/components/ListView/ListViewItem.js]: would be nice to add defaultProp for `onExpandClose` with default to `noop` so we can simplify the syntax above

Comment (priley86) [reply src/components/ListView/ListViewItem.js]: also missed in previous review... inside `toggleExpanded(index) {` method, the `index` is not used.

**Comment by priley86** [comment]
thanks @jeff-phillips-18,  it is nice addition that was missed before. other than comments above, i think it looks great!

**Comment by jeff-phillips-18** [comment]
Addressed the issues above and rebased.

File: src/components/ListView/ListViewItem.js [hunk 3]
```diff
@@ -39,18 +42,21 @@ class ListViewItem extends React.Component {
...
-            <ListViewExpand
-              expanded={expanded}
-              toggleExpanded={() => this.toggleExpanded()}
-            />
+      if (compoundExpand) {
+        return (
+          <ListViewGroupItem
+            expanded={compoundExpanded}
+            stacked={stacked}
+            {...other}
```
Comment (cdcabrera) [inline src/components/ListView/ListViewItem.js]: This looks like the issue related to the warning I was seeing... just need to filter these so they're specific to the child ListViewGroupItem.

**Comment by jeff-phillips-18** [comment]
Fixed the console errors.

Bumped patternfly version while I'm at it.

File: src/components/ListView/ListViewItem.js [hunk 4]
```diff
@@ -23,6 +23,7 @@ class ListViewItem extends React.Component {
 
   toggleExpanded() {
     const { onExpand, onExpandClose } = this.props;
+    console.dir(onExpand);
```
Comment (cdcabrera) [inline src/components/ListView/ListViewItem.js]: Missed this...

**Comment by cdcabrera** [comment]
@priley86 probably good to give another approve

**Comment by priley86** [comment]
@cdcabrera oops I hit merge instead ;)

**Pull request merged**